backend/src/api/database-migration.ts

```diff
@@ -1130,6 +1130,17 @@ class DatabaseMigration {
       await this.$executeQuery('ALTER TABLE blocks ADD INDEX `definition_hash` (`definition_hash`)');
       await this.updateToSchemaVersion(95);
     }
+
+    // Make definition_hash nullable
+    if (databaseSchemaVersion < 96) {
+      let poolJsonSha = '0138019f39e0f6033db576be15a4880a216c1dd0';
+      const [poolJsonShaDb]: any[] = await DB.query(`SELECT string FROM state WHERE name = 'pools_json_sha'`);
+      if (poolJsonShaDb?.length > 0) {
+        poolJsonSha = poolJsonShaDb[0].string;
+      }
+      await this.$executeQuery(`ALTER TABLE blocks ADD definition_hash varchar(255) NULL DEFAULT "${poolJsonSha}"`);
```

Comment:
doesn't the `definition_hash` column already exist from the previous migration?

Reply:
I'm also slightly confused about the default value here

Reply:
Yes sorry I've changed it to `MODIFY COLUMN`.
Default value is whatever is the current latest commit hash in https://github.com/mempool/mining-pools

Reply:
if we have (and set) a default, does it still need to be nullable? or is that used somewhere else?

Reply:
Originally the nullable fix was to make sure the block indexing does not get stuck if the `pools-v2.json` sha cannot be fetched from github.

I've then added the early exit check to make sure we don't even try in such case _(no sha in db AND no sha from github, so a first run without any pool definition)_.
https://github.com/mempool/mempool/blob/b5d5231bc1745eb834a0d4fbfe6799043152bc14/backend/src/index.ts#L154-L158

So in theory I don't think the nullable is still needed but I'm thinking there might be a weird config combination where this could actually be needed so I left it to make sure it does get the main loop stuck.

Reply:
Also maybe we should flip those two blocks actually to ensure we initialize `currentSha` before attempting to hit github

https://github.com/mempool/mempool/blob/f909d6bca52138131274b4dc3d1db28143514f3e/backend/src/tasks/pools-updater.ts#L48-L55